keepassxc-browser/content/autocomplete.js

```diff
@@ -106,25 +106,39 @@ kpxcAutocomplete.create = function(input, showListInstantly = false) {
         return list.getElementsByTagName('div');
     };
 
+    /*
+     * Keyboard shortcuts for autocomplete menu:
+     * - ArrowDown shows the list or selects item below, or the first item (last is active)
+     * - ArrowUp selects item above, or the last item (first is active)
+     * - Enter or Tab selects the item
+     * - Backspace and Delete shows the list if input field is empty. First item is activated
+    */
     function keyPress(e) {
         const item = getAllItems();
         if (e.key === 'ArrowDown') {
-            ++_index;
-            activateItem(item);
+            // If the list is now shown, show it
+            if (item.length === 0) {
+                _index = -1;
+                showList(input);
+            } else {
+                // Activate next item
+                ++_index;
+                activateItem(item);
+            }
         } else if (e.key === 'ArrowUp') {
-             --_index;
+            --_index;
             activateItem(item);
-        } else if (e.key === 'Enter') {
+        } else if (e.key === 'Enter' || e.key === 'Tab') {
             if (input.value === '') {
```

Comment:
Now this prevents <kbd>Tab</kbd>ing to the password field (or anywhere) when the input field is empty. I think we should only prevent the default action from being executed for <kbd>Enter</kbd>, which we could check here.

Reply:
Ah, true! Thanks for noticing.